mllib/src/main/scala/org/apache/spark/mllib/feature/ChiSqSelector.scala

```diff
@@ -189,11 +232,21 @@ class ChiSqSelector @Since("1.3.0") (
    */
   @Since("1.3.0")
   def fit(data: RDD[LabeledPoint]): ChiSqSelectorModel = {
-    val indices = Statistics.chiSqTest(data)
-      .zipWithIndex.sortBy { case (res, _) => -res.statistic }
-      .take(numTopFeatures)
-      .map { case (_, indices) => indices }
-      .sorted
+    val chiSqTestResult = Statistics.chiSqTest(data)
+    val features = selectorType match {
+      case ChiSqSelectorType.KBest => chiSqTestResult
+        .zipWithIndex.sortBy { case (res, _) => -res.statistic }
```

Comment:
Yeah it's true, though in general the result makes sense to sort. Then again I guess the model right now doesn't make use of that ordering. It might later. I don't feel strongly about it but seems simpler to always sort.